.ci/verify

```diff
@@ -1,19 +1,53 @@
-#!/bin/bash
+#!/bin/bash -x
 
 set -o errexit
 set -o nounset
 set -o pipefail
 
+concourse_cache_dir="$PWD/.cache_concourse"
+use_cache=no
+[ -d "$concourse_cache_dir" ] && use_cache=yes
+
 cd "$(dirname $0)/.."
 
 git config --global user.email "gardener@sap.com"
 git config --global user.name "Gardener CI/CD"
 
+# TODO: use pre-built image instead of installing in every pipeline run
 apt-get update
 apt-get install -y unzip jq parallel
 
 mkdir -p /go/src/github.com/gardener/gardener
 cp -r . /go/src/github.com/gardener/gardener
 cd /go/src/github.com/gardener/gardener
 
-make verify-extended
+if [ $use_cache = yes ] ; then
+  make install-requirements
+
+  # list of all to-be-cached directories
+  cache_dirs=(
+    "$(go env GOCACHE)"
+    "$(golangci-lint cache status | grep -i 'dir:' | awk '{print $2}')"
+    # TODO: cache tools (golangci-lint, mockgen, envtest bins, promtool, protoc, ...)
+  )
+
+  echo "> retrieving all cached dirs from concourse cache dir"
+  for cache in "${cache_dirs[@]}" ; do
+    mkdir -p "$cache"
+    cache_name=$(echo "$cache" | tr ' /' '_')
+    if [ -d "$concourse_cache_dir/$cache_name" ] ; then
+      cp -TR "$concourse_cache_dir/$cache_name" "$cache"
+    fi
+  done
+
+  # run test instead of test-cov to speed-up PR verification as `-cover` flag is not "cacheable"
+  make check-generate verify
+
+  echo "> moving all cached dirs back to concourse cache dir"
+  for cache in "${cache_dirs[@]}" ; do
+    cache_name=$(echo "$cache" | tr ' /' '_')
+    cp -TR "$cache" "$concourse_cache_dir/$cache_name"
```

Comment:
Unfortunately, `mv -T` failed during my tests with the following error:
```
mv: inter-device move failed: '/tmp/build/86a4261d/.cache_concourse/_root_.cache_go-build' to '/root/.cache/go-build'; unable to remove target: Directory not empty
```

So I switched to `cp -RT` instead. Though, this might grow the cache size infinitely over time, I think.

@ccwienk should we use a different mechanism instead?
You mentioned `rsync`, but I think we might also be able to achieve the same results with a `rm "$concourse_cache_dir/$cache_name"` before doing a `mv`, right?

Reply:
Ok, changed to `rm -rf` + `mv` now.
Have to give it another test spin in concourse for verifying that it works.
/hold

Reply:
Test run worked well: https://concourse.ci.gardener.cloud/teams/gardener/pipelines/gardener-master/jobs/master-pull-request-job/builds/613#L6072d300:622
So we can keep this approach :)
/unhold